include/benchmark/benchmark_api.h

```diff
@@ -858,6 +861,10 @@ class Fixture : public internal::Benchmark {
 #define BENCHMARK_MAIN()                   \
   int main(int argc, char** argv) {        \
     ::benchmark::Initialize(&argc, argv);  \
+    if (argc > 1) {                        \
```

Comment:
You could have `ReportUnrecognizedArguments(...)` return `true` if there are unrecognized arguments and false otherwise.  Then change this to `if (ReportUnrecognizedArguments(...)) exit(1);`